otel-extension/src/main/scala/io/scalac/mesmer/otelextension/instrumentations/zio/ConcurrentMetricRegistryPoller.scala

```diff
@@ -0,0 +1,60 @@
+package io.scalac.mesmer.otelextension.instrumentations.zio
+
+import java.util.Timer
+import java.util.TimerTask
+
+import zio.metrics.MetricKey
+import zio.metrics.MetricKeyType
+
+import scala.collection.mutable
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+import io.scalac.mesmer.otelextension.instrumentations.zio.ZIOMetrics._
+
+class ConcurrentMetricRegistryPoller {
+
+  private val timer = new Timer()
+
+  private val instruments = mutable.HashMap.empty[MetricKey[MetricKeyType], AutoCloseable]
+
+  private def task = new TimerTask {
+    override def run(): Unit =
+      try {
+        val snapshot = ConcurrentMetricRegistryClient.snapshot()
+
+        snapshot.filter { metricPair =>
+          !instruments.contains(metricPair.metricKey)
+        }.foreach { metricPair =>
+          val autocloseable = metricPair.metricKey.keyType match {
+            case _: MetricKeyType.Counter =>
```

Comment:
Is it possible to use the same OTEL async Instrument instance for ZIO metrics with the same name but different attr+value sets? There is a new API for that in OTEL (I think): 

```
    meter
      .counterBuilder(metricName(metricKey.name))
      .ofDoubles()
      .buildObserver()
 ```
 
 that allows for batch callbacks (callbacks for multiple measurements). 
 
 IMHO, if possible we should do it in a followup - just making sure if that makes sense

Reply:
I guess so, but signature does look funky there:
- https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/metrics/Meter.java#L141-L147
- https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/metrics/ObservableDoubleMeasurement.java
The signatures do not make their use very clear for me.

Also, this would also conflict with a polling-free approach tied to lifecycle of `MetricKey` that I explained here - https://github.com/ScalaConsultants/mesmer/pull/639#discussion_r1135223341.